pkg/diff/sql_vertex_generator.go

```diff
@@ -0,0 +1,229 @@
+package diff
+
+import (
+	"fmt"
+
+	"github.com/stripe/pg-schema-diff/internal/schema"
+)
+
+type diffType string
+
+const (
+	diffTypeDelete   diffType = "DELETE"
+	diffTypeAddAlter diffType = "ADDALTER"
+)
+
+// partialSQLGraph represents the SQL statements and their dependencies. Every SQL statement is a
+// vertex in the graph. This is different from a graph because a dependency may exist between a node
+// within this part and a node in another part.
+type partialSQLGraph struct {
+	vertices     []sqlVertex
+	dependencies []dependency
+}
+
+func (s *partialSQLGraph) statements() []Statement {
+	var statements []Statement
+	for _, vertex := range s.vertices {
+		statements = append(statements, vertex.statements...)
+	}
+	return statements
+}
+
+func concatPartialGraphs(parts ...partialSQLGraph) partialSQLGraph {
```

Comment:
thought: It might be nice to get rid of concatPartialGraphs and just have graphFromPartials take a list of partials. That way we could potentially add validation on partial construction (ie no dupe deps or vertices). It also might be nice to have each separated partial right up until the point where we merge them.

Reply:
I don't think we can get rid of `concatPartialGraphs` because there may be some "nested SQL generators", where we would have concat multiple partials into a bigger partial.
- TableSQLGenerator -> (ColumnSQLGenerator | CheckConstraintSQLGenerator | etc). Each returns a partial that would unioned, still be a partial, because they might have dependenciecs on higher level SQL generators

Reply:
One thing I like about concating partial graphs as we go is it follows a sort-of builder pattern. I think it's a bit more confusing if you have to carry local variables tens of lines down to be included in one function.

Reply:
I figure that sqlgenerators could return lists of partials, but yeah that's fair enough. I'm not too opinionated about this.